aws/cloudformation/components/database.yml.erb

```diff
@@ -16,19 +16,26 @@
     Type: AWS::RDS::DBInstance
     Properties:
       DBInstanceIdentifier: !Sub "${AWS::StackName}-<%=i%>"
+      # Resource name generated from db_parameters.yml.erb
+      # Use same ParameterGroup for writer and all readers so that any reader can be promoted to writer during a Failover.
+      DBParameterGroupName: !Ref AuroraWriterDBParameters
       DBClusterIdentifier: !Ref AuroraCluster
       DBInstanceClass: db.r4.large
       DBSubnetGroupName: !ImportValue VPC-DBSubnetGroup
       Engine: aurora-mysql
       # We will usually do engine version updates manually, so don't specify an EngineVersion for the DBInstance.
 <% end -%>
 
-  AuroraClusterDBParameters:
-    Type: AWS::RDS::DBClusterParameterGroup
+<%
+  YAML.load(erb_file(aws_dir('cloudformation/db_parameters.yml.erb'))).each do |key, values|
+-%>
+  <%=key%>DBParameters:
+    Type: AWS::RDS::DB<%=key.match('Cluster')%>ParameterGroup
     Properties:
-      Description: !Sub "Aurora DB Cluster Parameters for ${AWS::StackName}."
-      Family: aurora-mysql5.7
-      Parameters: {'innodb_monitor_enable': 'all'}
+      Description: !Sub "<%=key.titleize%> DB Parameters for ${AWS::StackName}."
+      Family: <%=key.include?('Aurora') ? 'aurora-' : ''%>mysql5.7
+      Parameters: <%=values.compact.to_json%>
+<%end -%>
```

Comment:
The `key.match` was a little confusing. I'm not sure what exactly the result is of this block. There are three keys in the file, one will create a `DBClusterParameterGroup`, the other two create `DBParameterGroup`s. If the key contains Aurora, then they `Family` will be set correctly. But all three contain `Aurora` so I'm not sure the point of that bit.

Reply:
Yes - I had trouble reverse engineering this and hadn't even noticed the `aurora-` logic. That's a legacy from when `db_parameters.yml.erb` supported generating both Aurora MySQL and old school RDS MySQL ParameterGroups. I'll remove this conditional and hard code to Aurora.